k8s-openapi-codegen-common/src/lib.rs

```diff
@@ -68,14 +70,33 @@ impl std::fmt::Display for Error {
 impl std::error::Error for Error {
 }
 
+lazy_static! {
+	pub static ref ROOT_K8S_OPENAPI: Vec<&'static str> = vec!["k8s_openapi"];
+}
+
+pub trait CrateRooter {
+	fn root(&self, namespace: &Vec<&str>) -> String;
+
+	fn for_namespace  (
+		&self,
+		namespace: &str
+	) -> String {
+		self.root(&namespace.split(r#"(::|\.)"#).collect())
```

Comment:
I don't understand what the `split` is doing. There is no instance of `(::|\.)` in the openshift openapi spec. It looks like a regex but `str::split` doesn't take regexes on stable (and even on nightly, it needs an actual `regex::Regex` object rather than a str literal.)

Reply:
That was indeed wrong. Fixed it and added a comment on the concept of `CrateRooter`.